app/models/usage_monitoring/alert.rb

```diff
@@ -0,0 +1,147 @@
+# frozen_string_literal: true
+
+module UsageMonitoring
+  class Alert < ApplicationRecord
+    include Discard::Model
+
+    self.discard_column = :deleted_at
+    self.inheritance_column = :alert_type
+
+    STI_MAPPING = {
+      "usage_amount" => "UsageMonitoring::UsageAmountAlert",
+      "billable_metric_usage_amount" => "UsageMonitoring::BillableMetricUsageAmountAlert"
+    }
+
+    CURRENT_USAGE_TYPES = %w[usage_amount billable_metric_usage_amount]
+    BILLABLE_METRIC_TYPES = %w[billable_metric_usage_amount]
+
+    default_scope -> { kept }
+
+    belongs_to :organization
+    belongs_to :billable_metric, optional: true
+
+    has_many :thresholds,
+      foreign_key: :usage_monitoring_alert_id,
+      class_name: "UsageMonitoring::AlertThreshold",
+      dependent: :delete_all
+
+    has_many :triggered_alerts,
+      foreign_key: :usage_monitoring_alert_id,
+      class_name: "UsageMonitoring::TriggeredAlert"
+
+    validate :billable_metric_when_required
+
+    def self.find_sti_class(type_name)
+      STI_MAPPING.fetch(type_name).constantize
+    end
+
+    def self.sti_name
+      STI_MAPPING.invert.fetch(name)
+    end
+
+    def subscription
+      @subscription ||= organization
+        .subscriptions
+        .active
+        .order(started_at: :desc)
+        .find_by(external_id: subscription_external_id)
+    end
+
+    def find_thresholds_crossed(current)
```

Comment:
I think this should be in the in the `ProcessAlertService`, it is coupled with the service logic and it is not really re-usable as it requires the `current_usage` to be injected

Reply:
Here only the current value (a number) is injected, no the `current_usage` object.

Reply:
Ok I see.
Not a blocker for me, but I have the feeling that since this method is only useful in a single service as part of its "core" purpose, the code should be there

Reply:
Noted! I'll see how it goes when I add the next two types of alerts